base/logger.go

```diff
@@ -34,3 +37,34 @@ func logCollationWorker(collateBuffer chan string, logger *log.Logger, bufferSiz
 		}
 	}
 }
+
+//runLogDeletion will delete rotated logs for the supplied logLevel. It will only perform these deletions when the
+//cumulative size of the logs are above the supplied sizeLimitMB.
+//logDirectory is the supplied directory where the logs are stored.
+func runLogDeletion(logDirectory string, logLevel string, sizeLimitMB int) {
+
+	sizeLimitMB = sizeLimitMB * 1024 * 1024 //Convert MB input to bytes
+	files, err := ioutil.ReadDir(logDirectory)
+
+	if err != nil {
+		Errorf(KeyAll, "Error reading log directory: %v", err)
+	}
+
+	totalSize := 0
+	indexDeletePoint := -1
+	for i := len(files) - 1; i >= 0; i-- {
```

Comment:
A high-level comment describing what this is doing would probably be helpful.

I.e. "Traverse backwards through sorted log filenames until we exceed the given size limit, and then remove any past that."